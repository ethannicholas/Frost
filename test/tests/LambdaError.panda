method test(f:(Int8)=>(Int8)):Int8 {
    return f(12)
}

method test(f:(Int16)=>(Int16)):Int16 {
    return f(12)
}

method test(f:(String)=>(String)):String {
    return f("Hello")
}

method main() {
    Console.printLine(test(x => x.utf8))
}
