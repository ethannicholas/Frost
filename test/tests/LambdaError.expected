LambdaError.panda:14:28: error: lambda error
    when interpreted as '(panda.core.Int8)=>(panda.core.Int8)':
        class 'panda.core.Int8' does not have a member named 'utf8'
    when interpreted as '(panda.core.String)=>(panda.core.String)':
        expected 'panda.core.String', but found 'panda.collections.ListView<panda.core.Char8>'
    when interpreted as '(panda.core.Int16)=>(panda.core.Int16)':
        class 'panda.core.Int16' does not have a member named 'utf8'
1 error
